### PR TITLE
Validate other_income_types

### PR DESCRIPTION
**Pull request by bengolder**
- Checks to ensure that any set of choices for the other_income_types
  array are a subset of predefined options
- Adds an error if not

### CODE REVIEW

File: app/models/member.rb [hunk 1]
```diff
@@ -49,12 +58,20 @@ class Member < ApplicationRecord
     inclusion: { in: SEXES },
     allow_nil: true
 
+  validate :correct_other_income_types
```
Comment (jessieay) [inline app/models/member.rb]: this is a "custom validator", which is a totally cool thing to do and in many cases necessary. In this case, rails gives us [a validator we can use](http://guides.rubyonrails.org/active_record_validations.html#inclusion):

```
 validates :other_income_types,
    inclusion: { in: OTHER_INCOME_TYPES },
    allow_nil: true

```
that "inclusion" validator will do the same thing that your custom validator is doing.

File: spec/models/dhs1426_pdf_spec.rb [hunk 2]
```diff
@@ -152,40 +163,54 @@
         primary_member_self_employed: "Yes",
         primary_member_first_employed_employer_name: "AA Accounting",
         primary_member_first_employed_pay_interval_hourly: "Yes",
-        # primary_member_first_employed_hours_interval_weekly: nil,
```
Comment (jessieay) [inline spec/models/dhs1426_pdf_spec.rb]: seems like these un-commenting changes are perhaps not related to this specific PR?

File: spec/models/member_spec.rb [hunk 3]
```diff
@@ -164,4 +164,36 @@
       end
     end
   end
+
+  describe "#other_income_types" do
```
Comment (jessieay) [inline spec/models/member_spec.rb]: to test this, we can use a cool library called `shoulda` (which we are already using) https://github.com/thoughtbot/shoulda-matchers/blob/master/lib/shoulda/matchers/active_model/validate_inclusion_of_matcher.rb

so we could replace all of these tests with:

```
it { should validate_inclusion_of(:other_income_tpes).in_array(Member::OTHER_INCOME_TYPES) }
it { should allow_value(nil).for(:other_income_types) }
```

File: spec/models/medicaid_application_member_attributes_spec.rb [hunk 4]
```diff
@@ -60,6 +69,13 @@
         primary_member_second_employed_employer_name: "BB Burgers",
         primary_member_second_employed_pay_interval_weekly: "Yes",
         primary_member_second_employed_pay_quantity: "222",
+        primary_member_additional_income_none: nil,
```
Comment (jessieay) [inline spec/models/medicaid_application_member_attributes_spec.rb]: this seems unrelated to this PR?

File: app/models/member.rb (hunk 1 above)
Comment (disambiguator) [reply app/models/member.rb]: The problem with the Rails validator here is that `other_income_types` is not a normal string but a magic array type.
`t.string "other_income_types", default: [], array: true`

I don't know if you can use the inclusion validator in this case sadly.

File: spec/models/medicaid_application_member_attributes_spec.rb (hunk 4 above)
Comment (disambiguator) [reply spec/models/medicaid_application_member_attributes_spec.rb]: Agree, there are a lot of different things happening. Hard to sift through what's what.
